crates/ruff_server/src/session/index.rs

```diff
@@ -347,6 +337,12 @@ impl Index {
     }
 
     pub(super) fn close_document(&mut self, key: &DocumentKey) -> crate::Result<()> {
+        if let DocumentKey::NotebookCell(uri) = key {
+            if self.notebook_cells.remove(uri).is_none() {
+                tracing::warn!("Tried to remove a notebook cell that does not exist: {uri}",);
+            }
+            return Ok(());
+        }
```

Comment:
Technically this is just moving over the code that removes the cell from the index 😄 

The error happened because we try to take a snapshot and that fails. This is a less critical error because if we fail to remove a notebook cell from the index, that means it's already been removed for one reason or another.